deckhouse-controller/pkg/controller/deckhouse-release/controller.go

```diff
@@ -551,6 +557,18 @@ func (r *deckhouseReleaseReconciler) tagUpdate(ctx context.Context, pods []corev
 	return nil
 }
 
+func getNewestPod(pods []corev1.Pod) corev1.Pod {
+	result := pods[0]
+
+	for _, pod := range pods[1:] {
```

Comment:
I guess we can filter a pod [here](https://github.com/deckhouse/deckhouse/pull/9621/files#diff-f5f1549dccbd7351a2389d6bc059b5f448e7e169efc48647e51bb86c2a1ec718R398). No need to return the full list and then filter it for the second time

Reply:
Why not return a full list of deckhouse pods? as I see it - we check all pods' statuses and cancel update if any pod isn't ready or in pending state.

Reply:
Why not return a full list of deckhouse pods? as I see it - we check all pods' statuses and cancel update if any pod isn't ready or in pending state.

Reply:
We need to figure out the image hash of the last running deckhouse pod, and decide, do we have to update or not.
I don't see the reason, why do we need the full list of deckhouse pods.